converters/fromZigbee.js

```diff
@@ -4370,6 +4370,26 @@ const converters = {
             return result;
         },
     },
+    xiaomi_switch_type: {
+        cluster: 'aqaraOpple',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            if (msg.data.hasOwnProperty(0x000A)) {
+                const lookup = {1: 'toggle', 2: 'momentary'};
+                return {switch_type: lookup[msg.data[0x000A]]};
+            }
+        },
+    },
+    xiaomi_switch_power_outage_memory: {
+        cluster: 'aqaraOpple',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            if (msg.data.hasOwnProperty(0x0201)) {
+                const lookup = {false: 'false', true: 'true'};
+                return {power_outage_memory: lookup[msg.data[0x0201]]};
```

Comment:
What does this converter return, a boolean or string? (should be booelean)

Reply:
BOOLEAN

Reply:
I don't understand it, if e.g. the value of `msg.data[0x0201]` is `false` (boolean) and you take `false` from the lookup the result is `"false"` (string)

Reply:
Sorry man, I'm not sure about it. I was copying it from another device. Basically, I need to get false or true value. Not sure how to do this.

Reply:
Can you provide the debug log when this report is recevied?

See https://www.zigbee2mqtt.io/information/debug.html on how to enable debug logging.

Reply:
```
Debug Received Zigbee message from '0x04cf8cdf3c8e92fe', type 'attributeReport', cluster 'genDeviceTempCfg', data '{"currentTemperature":23}' from endpoint 1 with groupID 0
Info MQTT publish: topic 'zigbee2mqtt/0x04cf8cdf3c8e92fe', payload '{"device_temperature":23,"linkquality":189}'
Debug Received MQTT message on 'zigbee2mqtt/0x04cf8cdf3c8e92fe/set' with data '{"power_outage_memory":true}'
Debug Publishing 'set' 'power_outage_memory' to '0x04cf8cdf3c8e92fe'
Info MQTT publish: topic 'zigbee2mqtt/0x04cf8cdf3c8e92fe', payload '{"device_temperature":23,"linkquality":189,"power_outage_memory":true}'
Debug Received Zigbee message from '0x04cf8cdf3c8e92fe', type 'attributeReport', cluster 'aqaraOpple', data '{"513":1}' from endpoint 1 with groupID 0
Info MQTT publish: topic 'zigbee2mqtt/0x04cf8cdf3c8e92fe', payload '{"device_temperature":23,"linkquality":189}'
Debug Received MQTT message on 'zigbee2mqtt/0x04cf8cdf3c8e92fe/set' with data '{"power_outage_memory":false}'
Debug Publishing 'set' 'power_outage_memory' to '0x04cf8cdf3c8e92fe'
Info MQTT publish: topic 'zigbee2mqtt/0x04cf8cdf3c8e92fe', payload '{"device_temperature":23,"linkquality":189,"power_outage_memory":false}'
Debug Received Zigbee message from '0x04cf8cdf3c8e92fe', type 'attributeReport', cluster 'aqaraOpple', data '{"513":0}' from endpoint 1 with groupID 0
Info MQTT publish: topic 'zigbee2mqtt/0x04cf8cdf3c8e92fe', payload '{"device_temperature":23,"linkquality":189}'
Debug Received MQTT message on 'zigbee2mqtt/0x04cf8cdf3c8e92fe/set' with data '{"power_outage_memory":true}'
Debug Publishing 'set' 'power_outage_memory' to '0x04cf8cdf3c8e92fe'
Debug Received Zigbee message from '0x04cf8cdf3c8e92fe', type 'attributeReport', cluster 'genDeviceTempCfg', data '{"currentTemperature":23}' from endpoint 1 with groupID 0
```

Reply:
I've added read support now. Can you open the frontend and click the refresh icon next to power_outage_memory and provide the logging of that?

Reply:
```
Debug Received MQTT message on 'zigbee2mqtt/0x04cf8cdf3c8e92fe/get' with data '{"switch_type":""}'
Debug Publishing get 'get' 'switch_type' to '0x04cf8cdf3c8e92fe'
Debug Received Zigbee message from '0x04cf8cdf3c8e92fe', type 'readResponse', cluster 'aqaraOpple', data '{"10":1}' from endpoint 1 with groupID 0
Info MQTT publish: topic 'zigbee2mqtt/0x04cf8cdf3c8e92fe', payload '{"device_temperature":18,"energy":0,"linkquality":186,"power":0,"state":"OFF","switch_type":"toggle"}'
Debug Received Zigbee message from '0x04cf8cdf3c8e92fe', type 'attributeReport', cluster 'genDeviceTempCfg', data '{"currentTemperature":19}' from endpoint 1 with groupID 0
Info MQTT publish: topic 'zigbee2mqtt/0x04cf8cdf3c8e92fe', payload '{"device_temperature":19,"energy":0,"linkquality":183,"power":0,"state":"OFF","switch_type":"toggle"}'
```

Reply:
Thanks ,updated the code, can you check if it is still ok when you do a get now?

Reply:
Thanks a lot. Tested. Working like a charm.